language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/ToJsonGenerator.scala

```diff
@@ -127,14 +127,33 @@ private[inner] object ToJsonGenerator {
     (methods, staticImports)
   }
 
-  def forKey(keyDamlType: Type)(implicit packagePrefixes: PackagePrefixes) =
-    MethodSpec
+  def forKey(keyDamlType: Type)(implicit packagePrefixes: PackagePrefixes) = {
+    val encoder = MethodSpec
       .methodBuilder("keyJsonEncoder")
       .addModifiers(Modifier.PUBLIC)
       .returns(classOf[JsonLfEncoder])
       .addStatement("return this.key.map($L).orElse(null)", encoderOf(keyDamlType))
       .build()
 
+    val toString = MethodSpec
+      .methodBuilder("keyToJson")
+      .addModifiers(Modifier.PUBLIC)
+      .addStatement("var enc = keyJsonEncoder()")
+      .addStatement("if (enc == null) return null")
+      .addStatement("var w = new $T()", classOf[StringWriter])
+      .beginControlFlow("try")
+      .addStatement("enc.encode(new $T(w))", classOf[JsonLfWriter])
+      .nextControlFlow("catch ($T e)", classOf[IOException])
+      .addComment("Not expected with StringWriter")
+      .addStatement("throw new $T(e)", classOf[UncheckedIOException])
+      .endControlFlow()
+      .addStatement("return w.toString()")
```

Comment:
I'm thinking of putting this logic into the `JsonLfEncoder` interface as a default method `intoString` so it doesn't need to be repeated so much.

Reply:
https://github.com/DACH-NY/canton/pull/16502